e2e/util/namespace.go

```diff
@@ -49,20 +75,56 @@ func DeleteNamespace(ctx types.Context, cluster types.Cluster, name string) erro
 		},
 	}
 
-	err := cluster.Client.Delete(ctx.Context(), ns)
+	managed, err := isManagedByRamenE2e(ctx, cluster, ns)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			log.Debugf("Namespace %q not found in cluster %q", name, cluster.Name)
+
+			return nil
+		}
+
+		return err
+	}
+
+	if !managed {
+		log.Debugf("Namespace %q in cluster %q is not managed by ramen-e2e, skipping deletion.",
+			name, cluster.Name)
+
+		return nil
+	}
+
+	err = cluster.Client.Delete(ctx.Context(), ns)
 	if err != nil {
 		if !k8serrors.IsNotFound(err) {
 			return err
 		}
 
 		log.Debugf("Namespace %q not found in cluster %q", name, cluster.Name)
-
-		return nil
 	}
 
 	return nil
 }
 
+// CreateAppNamespaces creates a namespace on both drclusters with ramen-e2e label.
+func CreateAppNamespaces(ctx types.Context, namespace string) error {
```

Comment:
CreateAppNamespace doet not hint that this is done all the managed clusters. Maybe rename these functions to CreateNamespaceOnMangedClusters(), DeleteNamespaceOnManagedClusters(), and WaitForNamespaceDeleteOnManagedClusters().